### PR TITLE
 migrated account fragment to jetpack compose

### LINKED ISSUE #1533: Migrate xml to Jetpack compose - bank/ui/AccountsFragment
**Summary:**

Summarize your issue here.

**Steps to reproduce:**

How can we reproduce again the issue?

**Expected behavior:**

What did you expect the app to do?

**Observed behavior:**

What did you see instead?  Describe your issue in detail here.

**Device and Android version:**

What make and model device (e.g., Samsung Galaxy S3) did you encounter this on?  What Android
version (e.g., Android 4.0 Ice Cream Sandwich or Android 6.0 Marshmallow) are you running?  Is it
 the stock
version from the manufacturer or a custom ROM?

**Screenshots:**

Can be created by pressing the Volume Down and Power Button at the same time on Android 4.0 and higher.

### PR DESCRIPTION
**Pull request by NiranjanNlc**
## Issue Fix
Fixes #1533 

## Screenshots
 

https://github.com/openMF/mobile-wallet/assets/25600880/06245532-ac08-4294-8cd5-8309e9cfbae9

## Description
 Migrated AccountFragments to jetpack compose screen 

##

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio. 

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by PratyushSingh07** [comment]
Hi @NiranjanNlc I think this UI is a bit different from the existing one . I would insist that you follow the original design

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt [hunk 1]
```diff
@@ -8,129 +8,42 @@ import android.view.View
...
-                            activity,
-                            BankAccountDetailActivity::class.java
-                        )
-                        intent.putExtra(
-                            Constants.BANK_ACCOUNT_DETAILS,
-                            mBankAccountsAdapter!!.getBankDetails(position)
-                        )
-                        intent.putExtra(Constants.INDEX, position)
-                        startActivityForResult(intent, BANK_ACCOUNT_DETAILS_REQUEST_CODE)
-                    }
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt:109]: and have you handled this case in the new compose UI ?

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt [hunk 2]
```diff
@@ -8,129 +8,42 @@ import android.view.View
...
-    var mBankAccountsAdapter: BankAccountsAdapter? = null
-
-    @JvmField
-    @BindView(R.id.linked_bank_account_text)
-    var linkedAccountsText: TextView? = null
-
-    @JvmField
-    @BindView(R.id.tv_empty_no_transaction_history_subtitle)
-    var tvTransactionsStateSubtitle: TextView? = null
+    lateinit var binding: FragmentAccountsBinding
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt:33]: can we get rid of binding ? with compose we shouldn't need this

Comment (NiranjanNlc) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt:33]: it shall go while all the component of FinanceFragment is converted into jetpack compose.

**Comment by NiranjanNlc** [comment]
> Hi @NiranjanNlc I think this UI is a bit different from the existing one . I would insist that you follow the original design



> Hi @NiranjanNlc I think this UI is a bit different from the existing one . I would insist that you follow the original design



> Hi @NiranjanNlc I think this UI is a bit different from the existing one . I would insist that you follow the original design

Its because of relative layout on the previous one and can you provide  me  some hint on how should i approach for the relative layout in jetpack compose .

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt (hunk 1 above)
Comment (NiranjanNlc) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt:109]: oops i missed this , would implement it on next commit along with your comment on  desighn and binding .
Thanks ....

**Comment by NiranjanNlc** [comment]
> Hi @NiranjanNlc I think this UI is a bit different from the existing one . I would insist that you follow the original design

Changed the desighn and implemented the onClickListener as you suggested .

Please get it reviewed again and provide me suggestion.

**Comment by NiranjanNlc** [comment]
@PratyushSingh07 ,
please go through the updated UI in PR description.
Hope it is upto the par.
Waiting for your review .

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.mifos.mobilewallet.model.domain.BankAccountDetails
+import org.mifos.mobilewallet.mifospay.R
+import org.mifos.mobilewallet.mifospay.bank.viewmodel.BankAccountsViewModel
+
+@Composable
+fun AccountScreen(viewModel: BankAccountsViewModel,
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt]: Code not formatted, Please format the code.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                  onAddAccountClicked: () -> Unit,
+                  onAccountClicked: (BankAccountDetails) -> Unit){
+    val bankAccountDetails = viewModel.bankAccountDetailsList
+    AccountDisplay(bankAccountDetails,onAddAccountClicked,onAccountClicked)
+}
+
+@Composable
+fun AccountDisplay(
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt]: Rename it with AccountScreenContent

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+                .align(Alignment.CenterHorizontally)
+                .padding(bottom = 128.dp),
+            onAddAccountClicked = onAddAccountClicked
+        )
+    }
+}
+
+@Preview
+@Composable
+fun AccountScreenPreview()  {
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt]: Empty bank details preview is missing

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+            "ANDHRA", "Rakesh anna ", "Telegana",
+            "1234567890", "Savings"
+        ),
+        BankAccountDetails(
+            "PNB", "luv Pro ", "Gujrat",
+            "1234567890", "Savings"
+        )
+    )
+    AccountDisplay(bankAccountDetails, {},{})
+//    AccountDisplay(bankAccountDetails = emptyList(),{})
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt]: remove this

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AddAccountChip.kt [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+            modifier = Modifier.padding(horizontal = 8.dp, vertical = 4.dp),
+            color = colorResource(id = R.color.colorPrimary),
+            textAlign = TextAlign.Center
+        )
+    }
+}
+
+@Preview
+@Composable
+fun AddAccountChipPreview() {
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AddAccountChip.kt:52]: wrap around the MifosTheme every preview function

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/ItemCasualkList.kt [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.mifos.mobilewallet.model.domain.BankAccountDetails
+import org.mifos.mobilewallet.mifospay.R
+
+@Composable
+fun ItemCausualList(
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/ItemCasualkList.kt]: ambiguous naming convention. what is Causual

Comment (therajanmaurya) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/ItemCasualkList.kt]: Write something related to feature

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/ItemCasualkList.kt [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+                )
+            }
+        }
+        Divider(modifier = Modifier.padding(start = 8.dp, end = 8.dp, top = 8.dp, bottom = 8.dp))
+    }
+}
+
+@Preview
+@Composable
+fun ComposeItemPreview() {
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/ItemCasualkList.kt]: Preview name should be function name + Preview - suffix

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/PlaceHolderStateLayout.kt [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import org.mifos.mobilewallet.mifospay.R
+@Composable
+fun PlaceholderStateLayout(
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/PlaceHolderStateLayout.kt]: ambiguous naming

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt [hunk 11]
```diff
@@ -8,127 +8,48 @@ import android.view.View
...
-            setupUi()
-        } else {
-            hideEmptyStateView()
-            mRvLinkedBankAccounts!!.visibility = View.VISIBLE
-            linkedAccountsText!!.visibility = View.VISIBLE
-            mBankAccountsAdapter!!.setData(bankAccountList)
+        binding= FragmentAccountsBinding.inflate(inflater, container, false)
+        ButterKnife.bind(this,binding.root)
+        binding.accountScreen.setContent {
+            AccountScreen(viewModel, ::addAccountClicked, ::onAccountClicked)
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt]: Don't need to pass the viewmodel init the ViewModel through hilt. 

See `CardsScreen` as an example.

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt (hunk 3 above)
Comment (NiranjanNlc) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt]: done by ctrl+alt+L

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt (hunk 4 above)
Comment (NiranjanNlc) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt]: changed

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt (hunk 5 above)
Comment (NiranjanNlc) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt]: created the new preview function for empty list

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt (hunk 6 above)
Comment (NiranjanNlc) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt]: done

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AddAccountChip.kt (hunk 7 above)
Comment (NiranjanNlc) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AddAccountChip.kt:52]: done

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/ItemCasualkList.kt (hunk 8 above)
Comment (NiranjanNlc) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/ItemCasualkList.kt]: changed ListItemWithImage ..done

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/PlaceHolderStateLayout.kt (hunk 10 above)
Comment (NiranjanNlc) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/PlaceHolderStateLayout.kt]: changed to EmptyStatePlaceHolder

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt (hunk 11 above)
Comment (NiranjanNlc) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt]: done

**Comment by PratyushSingh07** [comment]
closing this as completed by #1579

**Pull request closed without merging**